roles/irods/tasks/icat.yml

```diff
@@ -25,7 +25,7 @@
 
 - name: Make sure that the hostname is set correct FQDN name
   ansible.builtin.hostname:
-    name: '{{ irods_icat_fqdn }}'
+    name: '{{ hostvars[inventory_hostname]['fqdn'] }}'
```

Comment:
```suggestion
    name: "{{ hostvars[inventory_hostname][fqdn] }}"
```
Nested single quotes.

Reply:
Check, the first single quotes should be double, but the second single quotes around 'fqdn' should stay as this is a dict key named fqdn and not a variable fqdn.